module/input/SensorFilter/src/SensorFilter.cpp

```diff
@@ -87,6 +86,36 @@ namespace module::input {
         return s.str();
     }
```

Comment:
It feels like we should remove the old one? Unless I am misunderstanding, they do basically the exact same thing

Reply:
100% use the old one...

Reply:
I was going to do the one above but they need to take in different params since additional information is printed for the servos, which is why it probably wasn't implemented before for servo errors

Reply:
Copying the same function seems like badness

Reply:
I might not understand it.. but it looks like they take the same arguments just in different ways..

Reply:
Okay I will try make them the same

Reply:
I don't think I can merge them into one since they will pass different param types since they print different datatypes.

Reply:
I don't really see why we're making this a function.. but anyway.. if you really want to, I would modify the original errorString function to take the error value (the cm740 and servos return the same error packet afaik). And then just pass the function the servo id as the string
`                            if (error != RawSensors::Error::OK) {
                                NUClear::log<NUClear::WARN>(
                                    makeErrorString("Servo " + std::to_string(id + 1) + " ("
                                                        + std::to_string(static_cast<ServoID>(id)) + "):",
                                                    original,
                                                    error));
                            }`

Reply:
Passing original still won't work either, since `original` and `input` are different types. It doesn't really need to be a function it was just to make it easier to follow and broken up a bit.

Reply:
I think it’s pretty important that we decompose sensor filter into some smaller functions if we get chances like this. Atm it’s a huge monolith which is a marathon to try to understand. Decomposition into functions is a big help for situations like that.

regarding the issue of the different types, I get the problem now. I am wondering if we could make a string out of the servo name or id then use the old function which just took the string and flags perhaps? `shared/utility/input` has functions for getting the string name of a servo iirc - maybe we could use that?

Reply:
I agree with the decomposition, i think if that's the goal it should be done as a whole, I don't find it too complex having dealt with it for a long time. 

I like these kinds of "utility" functions to be general and not too repeating, i find having multiple functions to print the same set of error messages hard to swallow

Reply:
I just made a new issue about breaking down sensor filter (#729) and will get onto doing that while trying to find potential problems. Not sure what to do about this current decomposition.